src/julia_gc.c

```diff
@@ -502,6 +517,11 @@ void GapRootScanner(int full)
     // mark our Julia module (this contains references to our custom data
     // types, which thus also will not be collected prematurely)
     JMark(Module);
+    jl_task_t *task = JuliaTLS->current_task;
+    size_t size;
+    int tid;
+    StackBottomBags = (Bag *)jl_task_stack_buffer(task, &size, &tid);
+    StackBottomBags = (Bag *)((char *)StackBottomBags + size);
```

Comment:
Since we now always recompute `StackBottomBags` here anyway, is there any reason left to keep it as a global?

Also, I wonder: Why do we need to use `TryMarkRangeReverse` in `GapTaskScanner`, but here we get away with using `TryMarkRange`? Is that some kind of hidden assumption about `current_task` here, e.g. that it equals `root_task`? Some comments explaining this would be great!

Reply:
I wasn't yet sure if it is always good.

We use `TryMarkRange()` here because we *know* both ends of the stack for certain, not just one.

Reply:
What about my question regarding the hidden assumption about `current_task`?

Reply:
Oops, let me be clear. `current_task` is the task that is currently being executed and therefore has the stack that we are currently using. This allows us to get the "bottom" of the stack from the task state rather from what `InitBags()` was called with.